examples/v3/plugins/protobuf/test_provider.py

```diff
@@ -0,0 +1,247 @@
+
+    Returns:
+        Response containing protobuf-serialized Person data.
+
+    Raises:
+        HTTPException: If person is not found.
+    """
+    if MOCK_ADDRESS_BOOK is None:
+        raise HTTPException(status_code=404, detail="Person not found")
+
+    # Find person by ID
+    for person in MOCK_ADDRESS_BOOK.people:
+        if person.id == person_id:
+            # Serialize person to protobuf bytes
+            protobuf_data = person.SerializeToString()
+            return Response(
+                content=protobuf_data,
+                media_type="application/x-protobuf",
+            )
+
+    raise HTTPException(status_code=404, detail="Person not found")
+
+
+@pytest.fixture(scope="session")
+def server() -> Generator[str, None, None]:
+    """
+    Fixture to start the FastAPI server for testing.
+
+    Yields:
+        The URL of the running server.
+    """
+    assert PROVIDER_URL.host is not None
+    assert PROVIDER_URL.port is not None
+    server = Server(
+        uvicorn.Config(
+            app,
+            host=PROVIDER_URL.host,
+            port=PROVIDER_URL.port,
+        )
+    )
+    with server.run_in_thread() as url:
+        yield url
+
+
+def test_provider(server: str) -> None:
+    """
+    Test the protobuf provider against the consumer contract.
+
+    This test uses the Pact Verifier to replay the consumer's interactions
+    against the running provider service. It verifies that the provider
+    correctly handles protobuf serialization and responds appropriately
+    to both successful and error scenarios.
+
+    The test:
+
+    1.   Configures the Verifier with the protobuf plugin
+    2.   Points the verifier to the pact file generated by the consumer
+    3.   Sets up state handlers to prepare test data
+    4.   Verifies all interactions match the contract
+    """
+    pact_file = (
+        Path(__file__).parents[3] / "pacts" / "protobuf_consumer-protobuf_provider.json"
+    )
+
+    verifier = (
+        Verifier("protobuf_provider")
+        .add_transport(url=server)
+        .add_source(pact_file)
+        .state_handler(provider_state_handler, teardown=True)
+    )
+
+    verifier.verify()
+
+
+def provider_state_handler(
+    state: str,
+    action: Literal["setup", "teardown"],
+    parameters: dict[str, Any] | None = None,  # noqa: ARG001
+) -> None:
+    """
+    Handle provider state setup and teardown.
+
+    This function is called by Pact to set up the provider's internal state
+    before each interaction is replayed. It ensures that the provider has
+    the necessary data to respond correctly to the consumer's requests.
+
+    Args:
+        state:
+            The provider state name from the consumer test.
+
+        action:
+            Either `"setup"` or `"teardown"`.
+
+        parameters:
+            Additional parameters (not used in this example).
+    """
+    if action == "setup":
+        {
+            "person with ID 1 exists": setup_person_exists,
```

Comment:
can we have an example using provider state params here? as per previous comment